p2p/proto/consensus.proto

```diff
@@ -9,107 +9,146 @@ import "google/protobuf/timestamp.proto";
 // WIP - will change
 
 
-message BlockNumber {
+message BlockLocation {
     Hash   parent_hash = 1;
     uint64 height      = 2;
     uint64 fork_id     = 3;
 }
 
 enum Step {
-    Propose = 0;
-    Prevote = 1;
-    Precommit = 2;
+    Propose = 1;
+    Prevote = 2;
+    Precommit = 3;
 }
 
 message Vote {
     enum  VoteType {
-        Prevote   = 0;
-        Precommit = 1;
+        Prevote   = 1;
+        Precommit = 2;
     };
 
     // We use a type field to distinguish between prevotes and precommits instead of different
     // messages, to make sure the data, and therefore the signature, are unambiguous, since the
     // votes have the same fields.
-    VoteType           vote_type    = 1;
-    BlockNumber        block_number = 2;
-    uint32             round        = 3;
+    VoteType           vote_type      = 3;
+    BlockLocation      block_location = 4;
+    uint32             round          = 5;
     // This is optional since a vote can be NIL.
-    optional Hash      block_hash   = 4;
+    optional Hash      block_hash     = 6;
     // Identifies the voter.
-    Address            voter        = 5;
+    Address            voter          = 7;
     // Voter's signature of the above fields.
-    ConsensusSignature signature    = 6;
+    ConsensusSignature signature      = 8;
 }
 
 message ProposalInit {
-    BlockNumber               block_number   = 1;
+    BlockLocation             block_location = 1;
     uint32                    proposal_round = 2;
     optional uint32           valid_round    = 3;
     Address                   proposer       = 4;
-    google.protobuf.Timestamp timestamp      = 5;
```

Comment:
removed the timestamp?

Reply:
Yes as discussed this is content, not part of consensus